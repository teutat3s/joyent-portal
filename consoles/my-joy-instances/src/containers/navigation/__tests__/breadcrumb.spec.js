import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Breadcrumb from '../breadcrumb';
import Theme from '@mocks/theme';

it('renders <Breadcrumb /> without throwing', () => {
  expect(
    renderer
      .create(
        <Theme>
          <Breadcrumb />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders <Breadcrumb match /> without throwing', () => {
  const match = {
    params: {
      instance: 'name'
    }
  };

  expect(
    renderer
      .create(
        <Theme>
          <Breadcrumb match={match} />
        </Theme>
      )
      .toJSON()
  ).toMatchSnapshot();
});
