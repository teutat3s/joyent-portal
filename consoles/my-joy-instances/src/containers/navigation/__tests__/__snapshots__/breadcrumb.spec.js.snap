// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Breadcrumb /> without throwing 1`] = `
.c12 {
  margin-left: 0.375rem;
  margin-right: 0.375rem;
  margin-top: 1.125rem;
  margin-bottom: 1.125rem;
}

.c3 {
  padding-top: 0.375rem;
  padding-bottom: 0.375rem;
}

.c2 {
  margin-right: auto;
  margin-left: auto;
  padding-right: 1.875rem;
  padding-left: 1.875rem;
}

.c5 {
  margin-right: auto;
  margin-left: auto;
}

.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.625rem;
  margin-left: -0.625rem;
}

.c7 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.625rem;
  padding-left: 0.625rem;
  display: block;
}

.c1 {
  box-sizing: border-box;
  width: 100%;
  padding-left: 0;
  padding-right: 0;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
  max-width: 62.5rem;
}

.c10 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 600;
  line-height: 1.5rem;
  font-size: 0.9375rem;
}

.c9 {
  font-weight: normal;
  margin: 0;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8:first-child a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:last-child svg {
  display: none;
}

.c11 {
  cursor: pointer;
  font-weight: 400;
  color: rgb(59,70,204);
}

.c0 {
  border-bottom: 0.0625rem solid rgb(216,216,216);
}

@media only screen and (min-width:48em) {
  .c5 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c5 {
    width: 56rem;
  }
}

@media only screen and (min-width:75em) {
  .c5 {
    width: 59rem;
  }
}

@media only screen and (min-width:0em) {
  .c7 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
  }
}

@media only screen and (max-width:47.9375rem) {
  .c4 {
    padding-left: 0.375rem;
    padding-right: 0.375rem;
  }
}

<div
  className="c0 c1 c2"
>
  <div
    className="c3"
  >
    <div
      className="c4 c5"
    >
      <div
        className="c6"
        name="breadcrum"
      >
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <h4
              className="c9 c10"
              name="breadcrum-item"
            >
              <a
                className="c11"
                to="/"
              >
                <div
                  className="c12"
                >
                  Compute
                </div>
              </a>
            </h4>
            <svg
              height="6"
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }
              viewBox="0 0 9.6 6"
              width="9.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.6,1.12,8.24,0,4.8,3.5,1.36,0,0,1.12,4.8,6Z"
                fill="rgb(151, 151, 151)"
              />
            </svg>
          </div>
          <div
            className="c8"
          >
            <h4
              className="c9 c10"
              name="breadcrum-item"
            >
              <a
                className="c11"
                to="/instances"
              >
                <div
                  className="c12"
                >
                  Instances
                </div>
              </a>
            </h4>
            <svg
              height="6"
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }
              viewBox="0 0 9.6 6"
              width="9.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.6,1.12,8.24,0,4.8,3.5,1.36,0,0,1.12,4.8,6Z"
                fill="rgb(151, 151, 151)"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders <Breadcrumb match /> without throwing 1`] = `
.c12 {
  margin-left: 0.375rem;
  margin-right: 0.375rem;
  margin-top: 1.125rem;
  margin-bottom: 1.125rem;
}

.c3 {
  padding-top: 0.375rem;
  padding-bottom: 0.375rem;
}

.c2 {
  margin-right: auto;
  margin-left: auto;
  padding-right: 1.875rem;
  padding-left: 1.875rem;
}

.c5 {
  margin-right: auto;
  margin-left: auto;
}

.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.625rem;
  margin-left: -0.625rem;
}

.c7 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.625rem;
  padding-left: 0.625rem;
  display: block;
}

.c1 {
  box-sizing: border-box;
  width: 100%;
  padding-left: 0;
  padding-right: 0;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
  max-width: 62.5rem;
}

.c10 {
  margin: 0;
  color: rgb(73,73,73);
  font-weight: 600;
  line-height: 1.5rem;
  font-size: 0.9375rem;
}

.c9 {
  font-weight: normal;
  margin: 0;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8:first-child a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:last-child svg {
  display: none;
}

.c11 {
  cursor: pointer;
  font-weight: 400;
  color: rgb(59,70,204);
}

.c0 {
  border-bottom: 0.0625rem solid rgb(216,216,216);
}

@media only screen and (min-width:48em) {
  .c5 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c5 {
    width: 56rem;
  }
}

@media only screen and (min-width:75em) {
  .c5 {
    width: 59rem;
  }
}

@media only screen and (min-width:0em) {
  .c7 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
  }
}

@media only screen and (max-width:47.9375rem) {
  .c4 {
    padding-left: 0.375rem;
    padding-right: 0.375rem;
  }
}

<div
  className="c0 c1 c2"
>
  <div
    className="c3"
  >
    <div
      className="c4 c5"
    >
      <div
        className="c6"
        name="breadcrum"
      >
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <h4
              className="c9 c10"
              name="breadcrum-item"
            >
              <a
                className="c11"
                to="/"
              >
                <div
                  className="c12"
                >
                  Compute
                </div>
              </a>
            </h4>
            <svg
              height="6"
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }
              viewBox="0 0 9.6 6"
              width="9.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.6,1.12,8.24,0,4.8,3.5,1.36,0,0,1.12,4.8,6Z"
                fill="rgb(151, 151, 151)"
              />
            </svg>
          </div>
          <div
            className="c8"
          >
            <h4
              className="c9 c10"
              name="breadcrum-item"
            >
              <a
                className="c11"
                to="/instances"
              >
                <div
                  className="c12"
                >
                  Instances
                </div>
              </a>
            </h4>
            <svg
              height="6"
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }
              viewBox="0 0 9.6 6"
              width="9.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.6,1.12,8.24,0,4.8,3.5,1.36,0,0,1.12,4.8,6Z"
                fill="rgb(151, 151, 151)"
              />
            </svg>
          </div>
          <div
            className="c8"
          >
            <h4
              className="c9 c10"
              name="breadcrum-item"
            >
              <a
                className="c11"
                to="/instances/name"
              >
                <div
                  className="c12"
                >
                  name
                </div>
              </a>
            </h4>
            <svg
              height="6"
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }
              viewBox="0 0 9.6 6"
              width="9.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.6,1.12,8.24,0,4.8,3.5,1.36,0,0,1.12,4.8,6Z"
                fill="rgb(151, 151, 151)"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
